src/index.js

```diff
@@ -69,15 +69,16 @@ class Backend {
         }
 
         return response.text();
-      })
+      }, () => callback(`failed loading ${url}`, false))
       .then((data) => {
```

Comment:
```javascript
if (data === null) return;

try {
  callback(null, parse(data, null));
} catch (e) {
  // ...
}
```

What do you think of this instead?

Reply:
I like it because we can use an explicit `null` return value from the first `then` to indicate to the second `then` that:

1. The fetch failed
2. and that it should not try to parse the data or call the callback again.

The second then should only be called if the original fetch is ok and `response.text()` resolves.

Reply:
See 29dc6b8edfd31693397469559860e98d8cf796a3